### PR TITLE
DRILL-4237: Implement murmur3 for HashHelper

### PR DESCRIPTION
**Pull request by chunhui-shi**
Unit Test was removed from this commit. Test codes and the the test vector generator program will be included in functional tests in a separate checkin.

While implementing hash function instead of inventing a hash function, one thing need to emphasize is, in these well tested and widely accepted algorithms, all parameters and steps to shuffle are carefully designed and tested by the inventors and the community(e.g. smhasher). It is not just generating 'random' result from 'random' input. So each implementations of the algorithm should prove their complete correctness by testing with the same standard as I am doing here: cover all code paths(in murmurhash and xxhash cases, it is all possible input lengths that may go through different code paths, and sign/unsign cases).

### CODE REVIEW

**Comment by chunhui-shi** [comment]
Previous pull request and comments: https://github.com/apache/drill/pull/408

**Comment by chunhui-shi** [comment]
https://github.com/apache/drill/pull/430

File: exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/HashHelper.java [hunk 1]
```diff
@@ -60,4 +60,45 @@ public static int hash(ByteBuffer buf, int seed) {
...
+  public static long hash64(double val, long seed){
+    return MurmurHash3.hash64(val, (int)seed);
+  }
+
+  public static long hash64(long start, long end, DrillBuf buffer, long seed){
+    return MurmurHash3.hash64(start, end, buffer, (int)seed);
+  }
+
+  public static int hash32(double val, long seed) {
+    //return com.google.common.hash.Hashing.murmur3_128().hashLong(Double.doubleToLongBits(val)).asInt();
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/HashHelper.java]: Remove this line

File: exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/MurmurHash3.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  public static long hash64(double val, long seed){
+    return murmur3_64(Double.doubleToLongBits(val), (int)seed);
+  }
+
+  public static long hash64(long start, long end, DrillBuf buffer, long seed){
+    return murmur3_64(start, end, buffer, (int)seed);
+  }
+
+  public static int hash32(double val, long seed) {
+    //return com.google.common.hash.Hashing.murmur3_128().hashLong(Double.doubleToLongBits(val)).asInt();
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/MurmurHash3.java]: Remove..

**Comment by amansinha100** [comment]
+1  (I had already reviewed the previous #430

**Pull request closed without merging**